src/digital/DigitalButton.cpp

```diff
@@ -47,6 +47,29 @@ int DigitalButton::getId() {
     return m_pin;
 }
 
+/**
+ * @brief Invert input logic.
+ * 
+ * This function allows the user to dynamically change the logic while the program is running.
+ * Inverted logic means that if the input voltage level is high (e.g. 5 V DC), it will be treated as logic 0. 
+ * Therefore, a low voltage level (0 V DC) will be represented as log 1.
+ */
+void DigitalButton::invertInputLogic() {
+    m_buttonPressed = LOW ? HIGH : LOW;    
+}
+
+/**
+ * @brief Check if input logic is inverted.
+ *
+ * Inverted logic means that if the input voltage level is high (e.g. 5 V DC), it will be treated as logic 0. 
+ * Therefore, a low voltage level (0 V DC) will be represented as log 1.
+ *
+ * @return True if logic is inverted.
+ */
+bool DigitalButton::isInputLogicInverted() {
+    if ( m_buttonPressed = LOW ) {return true;}
```

Comment:
This is not an evaluation but an assignment instead. Probably a bug on your side.